lib/streamlit/elements/media_proto.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018-2019 Streamlit Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Python 2/3 compatibility
+from __future__ import print_function, division, unicode_literals, absolute_import
+from future.types import newbytes
+from streamlit.compatibility import setup_2_3_shims
+
+setup_2_3_shims(globals())
+
+import io
+import base64
+import re
+
+from validators import url
+
+from streamlit.proto import Video_pb2
+
+
+def _reshape_youtube_url(url):
+    """Return whether URL is any kind of YouTube embed or watch link.  If so,
+    reshape URL into an embed link suitable for use in an iframe.
+
+    If not a YouTube URL, return None.
+
+    Parameters
+    ----------
+    url : str, bytes
+    
+    Example
+    -------
+    >>> print(process_video_url('https://youtu.be/_T8LGqJtuGc'))
+
+    .. output::
+        https://www.youtube.com/embed/_T8LGqJtuGc
+    """
+    # Regular expression explained at https://regexr.com/4n2l2
+    # Covers any youtube URL (incl. shortlinks and embed links) and
+    # extracts its code.
+    re_youtube_url = re.compile(
+        "http(?:s?):\/\/(?:www\.)?youtu(?:be\.com|\.be)\/(?P<watch>(watch\?v=)|embed\/)?(?P<code>[\w\-\_]*)(&(amp;)?[\w\?=]*)?"
```

Comment:
Move this regex to a file-level constant, and break it for readability and to fit into the 80char limit:

```python
# Regular expression explained at https://regexr.com/4n2l2 Covers any youtube
# URL (incl. shortlinks and embed links) and extracts its code.
YOUTUBE_RE = re.compile(
      # Protocol
      "http(?:s?):\/\/"
      # Domain
      "(?:www\.)?youtu(?:be\.com|\.be)\/"
      # Path and query string
      "(?P<watch>(watch\?v=)|embed\/)?(?P<code>[\w\-\_]*)(&(amp;)?[\w\?=]*)?"
)
```